### PR TITLE
Implement `prepare_sweep()`

### PR DESCRIPTION
**Pull request by danielgranhao**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -284,6 +284,12 @@ pub struct OfferInfo {
     pub status: OfferStatus,
 }
 
+pub struct SweepInfo {
+    pub address: String,
+    pub onchain_fee_rate: u32,
+    pub onchain_fee_sat: u64,
```
Comment (dleutenegger) [inline src/lib.rs:290]: Thinking about the use case here, I think it would make more sense to return the `onchain_fee` as an Amount.

Comment (danielgranhao) [reply src/lib.rs:290]: Yes, completely missed that, thanks!

Comment (danielgranhao) [reply src/lib.rs]: Fixed in [a92a7aa](https://github.com/getlipa/lipa-lightning-lib/pull/765/commits/a92a7aa585676e17f96fe46ff63b31fe07fa098b)

File: examples/node/cli.rs [hunk 2]
```diff
@@ -824,7 +824,8 @@ fn payment_uuid(node: &LightningNode, words: &mut dyn Iterator<Item = &str>) ->
 
 fn sweep(node: &LightningNode, address: String) -> Result<String> {
     let fee_rate = node.query_onchain_fee_rate()?;
-    Ok(node.sweep(address.to_string(), fee_rate)?)
+    let sweep_info = node.prepare_sweep(address.to_string(), fee_rate)?;
```
[the remark this answers is not in the document]
Comment (danielgranhao) [reply examples/node/cli.rs]: Fixed in [a92a7aa](https://github.com/getlipa/lipa-lightning-lib/pull/765/commits/a92a7aa585676e17f96fe46ff63b31fe07fa098b)

**Review by dleutenegger — APPROVED** [review]

**Pull request merged**